src/kafka/base.rs

```diff
@@ -163,8 +163,31 @@ pub trait AlertProducer {
             )?;
             debug!(?total_messages);
 
-            // TODO: need to get the expected count
-            // TODO: finish the produce-or-not logic
+            // Count the number of Avro files in the data directory
+            let avro_count = count_files_in_dir(&self.data_directory(), Some(&["avro"]))?;
```

Comment:
One thing we haven't discussed is what to do when the topic exists and the data directory does not. There would be no way to verify the message count in the topic. Do we trust what's there, or do we start over?

Whatever the case, it's safe to say we don't want propagate a "directory not found" error here.

Reply:
Good catch! This only works because I've run it in the past on the same dates--the perils of caching...